docs/contributing/provider-design.md

```diff
@@ -0,0 +1,133 @@
+
+Some examples of functionality that is not expected in this provider:
+
+* Raw HTTP(S) handling. See the [Terraform HTTP Provider](https://registry.terraform.io/providers/hashicorp/http/latest) and [Terraform TLS Provider](https://registry.terraform.io/providers/hashicorp/tls/latest) instead.
+* Kubernetes resource management beyond the EKS service APIs. See the [Terraform Kubernetes Provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest) instead.
+* Active Directory or other protocol clients. See the [Terraform Active Directory Provider](https://registry.terraform.io/providers/hashicorp/ad/latest/docs) and other available provider instead.
+* Functionality that requires additional software beyond the Terraform AWS Provider to be installed on the host executing Terraform. This currently includes the AWS CLI. See the [Terraform External Provider](https://registry.terraform.io/providers/hashicorp/external/latest) and other available providers instead.
+
+## Infrastructure as Code Suitability
+
+While a goal of the provider is to gain as much resource coverage as possible, from a design perspective, it is necessary to be pragmatic that there are APIs or portions of APIs that may not be wholly compatible with infrastructure as code concepts and managing them with that methodology. In these cases, it is recommended to open an AWS Support case (potentially encouraging others to do the same) to either fix or enhance the AWS service API so components are more self-contained and compatible with these concepts. Sometimes these types of AWS Support cases can also provide some good insight into the AWS service expectations or API behaviors that may not be well documented.
+
+## Resource Type Considerations
+
+The general heuristic of when to create or not create a separate Terraform resource is if the AWS service API implements create, read, and delete operations for a component. Terraform resources work best as the smallest blocks of infrastructure, where configurations can build abstractions on top of these such as [Terraform Modules](https://www.terraform.io/docs/modules/). However, not all AWS service API functionality falls cleanly into components to be managed with those types of operations. It may be necessary to consider other patterns of mapping API operations to Terraform resources and operations. This section highlights these design recommendations, such as when to consider implemantations within the same Terraform resource or as a separate Terraform resource.
```

Comment:
```suggestion
The general heuristic of whether or not to create a separate Terraform resource is if the AWS service API implements create, read, and delete operations for a component. Terraform resources work best as the smallest blocks of infrastructure, where configurations can build abstractions on top of these such as [Terraform Modules](https://www.terraform.io/docs/modules/). However, not all AWS service API functionality falls cleanly into components to be managed with those types of operations. It may be necessary to consider other patterns of mapping API operations to Terraform resources and operations. This section highlights these design recommendations, such as when to consider implemantations within the same Terraform resource or as a separate Terraform resource.
```